apps/greeter/mix.exs

```diff
@@ -0,0 +1,42 @@
+defmodule Greeter.Mixfile do
+  use Mix.Project
+
+  def project do
+    [app: :greeter,
+     version: "0.0.1",
+     build_path: "../../_build",
+     config_path: "../../config/config.exs",
+     deps_path: "../../deps",
+     lockfile: "../../mix.lock",
+     elixir: "~> 1.2",
+     elixirc_paths: elixirc_paths(Mix.env),
+     compilers: [:phoenix, :gettext] ++ Mix.compilers,
+     build_embedded: Mix.env == :prod,
+     start_permanent: Mix.env == :prod,
+     deps: deps()]
+  end
+
+  # Configuration for the OTP application.
+  #
+  # Type `mix help compile.app` for more information.
+  def application do
+    [mod: {Greeter, []},
+     applications: [:phoenix, :phoenix_pubsub, :phoenix_html, :cowboy, :logger, :gettext]]
+  end
+
+  # Specifies which paths to compile per environment.
+  defp elixirc_paths(:test), do: ["lib", "web", "test/support"]
+  defp elixirc_paths(_),     do: ["lib", "web"]
+
+  # Specifies your project dependencies.
+  #
+  # Type `mix help deps` for examples and options.
+  defp deps do
```

Comment:
I understand... I think I'm visualising something similar to the umbrella app in the Programming Phoenix book where apparently there is only just 1 Phoenix app. Perhaps we can think of Phoenix as the **web layer** for all of the other apps under the umbrella, being only that its responsibility. The other apps can have their own responsibility and rely no the Phoenix app in the umbrella so they can be reached from the _Internet_.